apps/api/src/app/events/services/direct-service/handlers/slack.handler.ts

```diff
@@ -0,0 +1,51 @@
+import { ICredentials, IChannelCredentials } from '@novu/dal';
+import { ChannelTypeEnum, IDirectOptions, IDirectProvider, ISendMessageSuccessResponse } from '@novu/stateless';
+import { BaseDirectHandler } from './base.handler';
+import axios from 'axios';
+
+export class SlackHandler extends BaseDirectHandler {
+  constructor() {
+    super('slack', ChannelTypeEnum.DIRECT);
+  }
+
+  buildProvider(credentials: ICredentials) {
+    const config: {
+      applicationId: string;
+    } = { applicationId: credentials.applicationId };
+
+    this.provider = new SlackProvider(config);
+  }
+
+  setSubscriberCredentials(credentials) {
+    this.provider.setSubscriberCredentials(credentials);
+  }
+}
+
+export class SlackProvider implements IDirectProvider {
```

Comment:
We should move this entire class to the providers folder

Reply:
Resolved in 9c076f92f5e62b2926f45a30b92b38135fe0b227